### ISSUE TITLE
More FAQs

### ISSUE THREAD

**Issue by brson** [rust-lang/rust-www#248]
There are still FAQs not answered:

* "What does the name 'Rust' mean?"
* "Why does a tiny program like "Hello World" produce such a large executable file?" - [answer](https://users.rust-lang.org/t/why-do-rust-programs-use-more-memory-than-the-c-haskell-and-ocaml-versions/3224/1).

**Comment by antoyo** [comment]
[The second question](https://users.rust-lang.org/t/why-do-rust-programs-use-more-memory-than-the-c-haskell-and-ocaml-versions/3224/1) is about RAM usage, not executable size.

**Comment by brson** [comment]
Comparison to swift would be useful.

**Comment by Manishearth** [comment]
@antoyo Yes, but the additional RAM usage is _caused_ by a larger executable (which is loaded into RAM).

For swift I guess something along the lines of it having a very similar type system without the ownership/borrowing, and using automatic reference counting for memory safety?

**Comment by antoyo** [comment]
@Manishearth In the thread on rust-lang, I learned that using the system allocator lower the RAM usage. Does this mean that jemalloc is using 10 MB of RAM?

**Comment by Manishearth** [comment]
That could be the case, but I doubt it; it probably means you're statically linking to jemalloc which increases the binary size (whereas the system allocator exists on your machine anyway)

**Comment by brson** [comment]
"What's the point of inner attributes?" / "Why are there so many attribute forms?" / "Why are doc comments attributes?"

**Comment by petrochenkov** [comment]
"Why does my Windows application show console?"

**Comment by kud1ing** [comment]
"Which string type should i use?"

http://hermanradtke.com/2015/05/03/string-vs-str-in-rust-functions.html

**Comment by kud1ing** [comment]
"Can i accept both `&str` and `String`?"

http://hermanradtke.com/2015/05/06/creating-a-rust-function-that-accepts-string-or-str.html

**Comment by F001** [comment]
Why there is no incremental/decremental operators ( ++/-- ) in Rust? 

Well, I know these operators lead to troubles, but I can't find the canonical explanation.

**Comment by AndrewBrinker** [comment]
I am happy to take a crack at answers to the above-listed questions. I can probably whip up reasonable answers to all of them but the naming question. If anyone familiar with the history has a good explanation for the name, I'm happy to write up an answer that explains it.
